lib/fsm/fsm.go

```diff
@@ -620,6 +606,8 @@ type StateChange struct {
 	State string
 	// Error is the error that happened during phase execution
 	Error trace.Error
+	// created overrides the time for the state change, only used in tests.
+	created time.Time
```

Comment:
So just by looking at the way this is used, I don't think you need this in the main code if that's only for tests. There's nothing special about [testEngine](https://github.com/gravitational/gravity/blob/roman/5.5/resume/lib/fsm/testhelpers.go#L33). More over, I think it's confusing that the test engine is implementing an additional functionality (as opposed to existing main code counterparts) just to relay a test attribute.
I think for your purpose, you could implement an additional test-only API:
```go
// ChangePhaseState records the provided phase state change in the test engine.
func (t *testEngine) ChangePhaseState(ctx context.Context, ch StateChange) error {
	t.changelog = append(t.changelog, storage.PlanChange{
		PhaseID:  ch.Phase,
		NewState: ch.State,
		Created:  t.clock.Now().Add(time.Duration(len(t.changelog)) * time.Minute),
	})
	return nil
}
...
func (t *testEngine) changePhaseStateWithTimestamp(ctx context.Context, ch StateChange, created time.Time) error {
	t.changelog = append(t.changelog, storage.PlanChange{
		PhaseID:  ch.Phase,
		NewState: ch.State,
		Created:  created,
	})
	return nil
}
```
and use `changePhaseStateWithTimestamp` where necessary.